src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs

```diff
@@ -1422,20 +1422,22 @@ private void ExecuteUninstallPackageCommand(object sender, ExecutedRoutedEventAr
                 return;
             }
 
-            UninstallPackage(package.Id);
+            UninstallPackage(package);
         }
 
-        private void SetOptions(NuGetUI nugetUi, NuGetActionType actionType)
+        private void SetOptions(NuGetUI nugetUi, NuGetActionType actionType, IEnumerable<PackageItemViewModel> packages)
         {
             var options = _detailModel.Options;
+            var vulnerablePkgs = packages.Where(x => x.Vulnerabilities.Any());
 
             nugetUi.FileConflictAction = options.SelectedFileConflictAction.Action;
             nugetUi.DependencyBehavior = options.SelectedDependencyBehavior.Behavior;
             nugetUi.RemoveDependencies = options.RemoveDependencies;
             nugetUi.ForceRemove = options.ForceRemove;
             nugetUi.DisplayPreviewWindow = options.ShowPreviewWindow;
             nugetUi.DisplayDeprecatedFrameworkWindow = options.ShowDeprecatedFrameworkWindow;
-
+            nugetUi.TopLevelVulnerablePackagesCount = vulnerablePkgs.Count();
+            nugetUi.TopLevelVulnerablePackagesMaxSeverity = vulnerablePkgs.SelectMany(x => x.Vulnerabilities).Max(y => y.Severity);
```

Comment:
Please replace `x` and `y` with names. eg, `packages`, 'vulnerability', or similar

Reply:
Fixed.